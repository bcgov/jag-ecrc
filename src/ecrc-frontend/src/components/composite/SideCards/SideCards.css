p {
  display: flex;
  flex-direction: column;
}
p > button {
  align-self: flex-end;
}

.dark-button {
  background-color: #2d476f;
  border-color: #294266;
  margin-right: 0px;
}

.block {
  display: contents;
}
