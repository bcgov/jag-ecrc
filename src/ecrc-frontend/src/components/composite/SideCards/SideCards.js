/* eslint-disable react/jsx-curly-newline */
import React from "react";
import PropTypes from "prop-types";

import "./SideCards.css";
import SideCard from "../../base/sideCard/SideCard";
import { Button } from "../../base/button/Button";
import MenuItem from "../../base/menuItem/MenuItem";

export default function SideCards({ type, sideCardLinks }) {
  // ACCESS CODE SIDECARD
  const accessCode = {
    heading: "Need help with an access code?",
    content: [
      "Access code is specific to each organization. If you need help with the organization access code, please ",
      <b className="block">contact your organization.</b>
    ],
    type: "bluegrey"
  };

  // BCSC SIDECARD
  const bcscbutton = {
    label: "READ MORE",
    buttonStyle: "btn btn-primary dark-button",
    buttonSize: "btn btn-md",
    type: "submit"
  };

  const bcservicesButton = (
    <Button
      key="bc-services-card-link"
      button={bcscbutton}
      onClick={() =>
        window.open(
          "https://www2.gov.bc.ca/gov/content/governments/government-id/bc-services-card"
        )
      }
    />
  );

  const bcservice = {
    heading: "Get a BC Services Card",
    content: [
      "B.C. residents who have lived in the province for at least six months must use a BC Services Card to log in to the online qualification tool. Learn how to get a card.",
      bcservicesButton
    ],
    type: "blue"
  };

  // CONTACT INFORMATION SIDECARD
  const contact = {
    heading: "Contact Information",
    content: [
      "Criminal Records Review Program staff are available Monday to Friday from 8:30 a.m. - 4:30 p.m.",
      <b>Office:</b>,
      "Toll free - 1 855 587-0185 (press option 2)",
      <b>Email:</b>,
      "criminalrecords@gov.bc.ca"
    ],
    type: "bluegrey"
  };

  // CRIMINAL RECORD SIDECARD
  const crbutton = {
    label: "VISIT THE CRIMINAL RECORD REVIEW WEBSITE",
    buttonStyle: "btn btn-primary dark-button",
    buttonSize: "btn btn-md",
    type: "submit"
  };

  const criminalRecordButton = (
    <Button
      key="criminal-record-link"
      button={crbutton}
      onClick={() =>
        window.open(
          "https://www2.gov.bc.ca/gov/content/safety/crime-prevention/criminal-record-check"
        )
      }
    />
  );

  const criminalRecord = {
    heading: "Why do I need to apply and consent to a criminal record check?",
    content: ["Learn more about the process", criminalRecordButton],
    type: "blue"
  };

  // USEFUL LINKS SIDECARD
  const links = sideCardLinks.map(sideCardLink => {
    return <MenuItem key={sideCardLink.name} menuItem={sideCardLink} />;
  });

  const usefulLinks = {
    heading: "Useful Links",
    content: [links],
    type: "blue"
  };

  return (
    <div>
      {type === "accesscode" && (
        <SideCard key="accesscode" sideCard={accessCode} />
      )}
      {type === "bcservice" && (
        <SideCard key="bcservices" sideCard={bcservice} />
      )}
      {type === "contactinformation" && (
        <SideCard key="contactinformation" sideCard={contact} />
      )}
      {type === "criminalrecord" && (
        <SideCard key="criminalrecord" sideCard={criminalRecord} />
      )}
      {type === "usefullinks" && (
        <SideCard key="usefullink" sideCard={usefulLinks} />
      )}
    </div>
  );
}

SideCards.propTypes = {
  type: PropTypes.string.isRequired,
  sideCardLinks: PropTypes.arrayOf(
    PropTypes.shape({
      name: PropTypes.string.isRequired,
      url: PropTypes.string.isRequired
    })
  )
};

SideCards.defaultProps = {
  sideCardLinks: []
};
