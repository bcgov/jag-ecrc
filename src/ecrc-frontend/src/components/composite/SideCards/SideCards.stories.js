import React from "react";

import SideCards from "./SideCards";

export default {
  title: "SideCards",
  component: SideCards
};

const links = [
  {
    name: "Home",
    url: "/"
  },
  {
    name: "Somewhere",
    url: "/somewhere"
  },
  {
    name: "Somewhere else",
    url: "/somewhereelse"
  },
  {
    name: "Somewhere with a long name for no reason",
    url: "/here"
  }
];

export const AccessCode = () => <SideCards type={"accesscode"} />;
export const BcService = () => <SideCards type={"bcservice"} />;
export const ContactInformation = () => (
  <SideCards type={"contactinformation"} />
);
export const CriminalRecord = () => <SideCards type={"criminalrecord"} />;
export const UsefulLinks = () => (
  <SideCards type={"usefullinks"} sideCardLinks={links} />
);
