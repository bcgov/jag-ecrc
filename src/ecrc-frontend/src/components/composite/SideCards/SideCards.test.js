import React from "react";
import { create } from "react-test-renderer";

import SideCards from "./SideCards";

describe("SideCards Component", () => {
  const links = [
    {
      name: "Home",
      url: "/"
    },
    {
      name: "Somewhere",
      url: "/somewhere"
    },
    {
      name: "Somewhere else",
      url: "/somewhereelse"
    },
    {
      name: "Somewhere with a long name for no reason",
      url: "/here"
    }
  ];

  test("AccessCode sidecard", () => {
    const accessCode = create(<SideCards type={"accesscode"} />);
    expect(accessCode.toJSON()).toMatchSnapshot();
  });

  test("BcServices sidecard", () => {
    const bcServices = create(<SideCards type={"bcservice"} />);
    expect(bcServices.toJSON()).toMatchSnapshot();
  });

  test("ContactInformation sidecard", () => {
    const contactInformation = create(
      <SideCards type={"contactinformation"} />
    );
    expect(contactInformation.toJSON()).toMatchSnapshot();
  });

  test("CriminalRecord sidecard", () => {
    const criminalRecord = create(<SideCards type={"criminalrecord"} />);
    expect(criminalRecord.toJSON()).toMatchSnapshot();
  });

  test("AccessCode sidecard", () => {
    const usefulLink = create(
      <SideCards type={"usefullinks"} sideCardLinks={links} />
    );
    expect(usefulLink.toJSON()).toMatchSnapshot();
  });
});
