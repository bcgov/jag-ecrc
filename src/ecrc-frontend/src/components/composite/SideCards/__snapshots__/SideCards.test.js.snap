// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SideCards Component AccessCode sidecard 1`] = `
<div>
  <div
    className="dashboard-spacing"
    style={
      Object {
        "position": "relative",
      }
    }
  >
    <div
      className="row"
    >
      <div
        className="col-12"
        style={
          Object {
            "paddingTop": "30px",
            "position": "relative",
          }
        }
      >
        <p />
        <section
          className="bluegrey-container"
          id="bluegrey-section"
        >
          <div
            className="container-background bluegrey-heading"
          >
            <h2
              className="heading-style"
            >
              Need help with an access code?
            </h2>
          </div>
          <div
            className="bluegrey-content"
          >
            <p
              className="content-style"
            >
              Access code is specific to each organization. If you need help with the organization access code, please 
              <b
                className="block"
              >
                contact your organization.
              </b>
            </p>
          </div>
        </section>
      </div>
    </div>
  </div>
</div>
`;

exports[`SideCards Component AccessCode sidecard 2`] = `
<div>
  <div
    className="dashboard-spacing"
    style={
      Object {
        "position": "relative",
      }
    }
  >
    <div
      className="row"
    >
      <div
        className="col-12"
        style={
          Object {
            "paddingTop": "30px",
            "position": "relative",
          }
        }
      >
        <p />
        <section
          className="submit-container"
          id="blue-section"
        >
          <h2
            className="heading-style"
          >
            Useful Links
          </h2>
          <div
            className="submit-content"
          >
            <p>
              <li
                className="menuItem"
              >
                <a
                  href="/"
                >
                  Home
                </a>
              </li>
              <li
                className="menuItem"
              >
                <a
                  href="/somewhere"
                >
                  Somewhere
                </a>
              </li>
              <li
                className="menuItem"
              >
                <a
                  href="/somewhereelse"
                >
                  Somewhere else
                </a>
              </li>
              <li
                className="menuItem"
              >
                <a
                  href="/here"
                >
                  Somewhere with a long name for no reason
                </a>
              </li>
            </p>
          </div>
        </section>
      </div>
    </div>
  </div>
</div>
`;

exports[`SideCards Component BcServices sidecard 1`] = `
<div>
  <div
    className="dashboard-spacing"
    style={
      Object {
        "position": "relative",
      }
    }
  >
    <div
      className="row"
    >
      <div
        className="col-12"
        style={
          Object {
            "paddingTop": "30px",
            "position": "relative",
          }
        }
      >
        <p />
        <section
          className="submit-container"
          id="blue-section"
        >
          <h2
            className="heading-style"
          >
            Get a BC Services Card
          </h2>
          <div
            className="submit-content"
          >
            <p>
              B.C. residents who have lived in the province for at least six months must use a BC Services Card to log in to the online qualification tool. Learn how to get a card.
              <button
                className="btn btn-primary dark-button btn btn-md"
                onClick={[Function]}
                type="submit"
              >
                READ MORE
              </button>
            </p>
          </div>
        </section>
      </div>
    </div>
  </div>
</div>
`;

exports[`SideCards Component ContactInformation sidecard 1`] = `
<div>
  <div
    className="dashboard-spacing"
    style={
      Object {
        "position": "relative",
      }
    }
  >
    <div
      className="row"
    >
      <div
        className="col-12"
        style={
          Object {
            "paddingTop": "30px",
            "position": "relative",
          }
        }
      >
        <p />
        <section
          className="bluegrey-container"
          id="bluegrey-section"
        >
          <div
            className="container-background bluegrey-heading"
          >
            <h2
              className="heading-style"
            >
              Contact Information
            </h2>
          </div>
          <div
            className="bluegrey-content"
          >
            <p
              className="content-style"
            >
              Criminal Records Review Program staff are available Monday to Friday from 8:30 a.m. - 4:30 p.m.
              <b>
                Office:
              </b>
              Toll free - 1 855 587-0185 (press option 2)
              <b>
                Email:
              </b>
              criminalrecords@gov.bc.ca
            </p>
          </div>
        </section>
      </div>
    </div>
  </div>
</div>
`;

exports[`SideCards Component CriminalRecord sidecard 1`] = `
<div>
  <div
    className="dashboard-spacing"
    style={
      Object {
        "position": "relative",
      }
    }
  >
    <div
      className="row"
    >
      <div
        className="col-12"
        style={
          Object {
            "paddingTop": "30px",
            "position": "relative",
          }
        }
      >
        <p />
        <section
          className="submit-container"
          id="blue-section"
        >
          <h2
            className="heading-style"
          >
            Why do I need to apply and consent to a criminal record check?
          </h2>
          <div
            className="submit-content"
          >
            <p>
              Learn more about the process
              <button
                className="btn btn-primary dark-button btn btn-md"
                onClick={[Function]}
                type="submit"
              >
                VISIT THE CRIMINAL RECORD REVIEW WEBSITE
              </button>
            </p>
          </div>
        </section>
      </div>
    </div>
  </div>
</div>
`;
